### PR TITLE
AK-27551 Implemented Azure ExpressRoute Connector

### PR DESCRIPTION
**Pull request by Alshadex**

### CODE REVIEW

File: examples/resources/alkira_connector_azure_er/resource.tf [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    size =              "LARGE"
+    enabled =           true
+    vhub_prefix =       "10.129.0.0/23"
+    cxp =               "USWEST-AZURE-1"
+    tunnel_protocol =   "VXLAN_GPE"
+    group =             alkira_group.tftest.name
+
+    # You can add more instances blocks.
+    instances {
+        name =                      "instance13"
```
Comment (spikewang) [inline examples/resources/alkira_connector_azure_er/resource.tf:12]: you probably want to run `terraform format` on this example to make it aligned properly.

File: alkira/resource_alkira_connector_azure_er.go [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+	"fmt"
+
+	"github.com/alkiranet/alkira-client-go/alkira"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func resourceAlkiraConnectorAzureEr() *schema.Resource {
+	return &schema.Resource{
+		Description: "Manage Azure ExpressRouter Connector.",
```
Comment (spikewang) [inline alkira/resource_alkira_connector_azure_er.go:13]: ExpressRouter -> ExpressRoute :)

File: alkira/resource_alkira_connector_azure_er.go [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+		Delete: resourceConnectorAzureErDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Description: "The name of the connector.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"credential_id": {
+				Type:     schema.TypeString,
```
Comment (spikewang) [inline alkira/resource_alkira_connector_azure_er.go:27]: Description is gone? I think this one will share the `credential_azure_vnet`?

File: alkira/resource_alkira_connector_azure_er.go [hunk 4]
```diff
@@ -0,0 +1,294 @@
...
+// resourceConnectorAzureErCreate create an Azure ExpressRoute connector
+func resourceConnectorAzureErCreate(d *schema.ResourceData, m interface{}) error {
+	client := m.(*alkira.AlkiraClient)
+	connector, err := generateConnectorAzureErRequest(d, m)
+
+	if err != nil {
+		return err
+	}
+
+	id, err := client.CreateConnectorAzureEr(connector)
```
Comment (spikewang) [inline alkira/resource_alkira_connector_azure_er.go:174]: The client change was not here yet? Guess the original function name is a bit confusing...

File: docs/resources/connector_azure_expressroute.md [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+  enabled         = true
+  vhub_prefix     = "10.129.0.0/23"
+  cxp             = "USWEST-AZURE-1"
+  tunnel_protocol = "VXLAN_GPE"
+  group           = alkira_group.tftest.name
+
+  # You can add more instances blocks.
+  instances {
+    name                     = "instance13"
+    express_route_circuit_id = "/subscriptions/45060700-1949-4d0f-ba2c-4241274e8fa1/resourceGroups/robin-test/providers/Microsoft.Network/expressRouteCircuits/er-automation"
```
Comment (spikewang) [inline docs/resources/connector_azure_expressroute.md:28]: could we remove this real subscriptions...

File: alkira/resource_alkira_connector_azure_er.go (hunk 3 above)
Comment (Alshadex) [reply alkira/resource_alkira_connector_azure_er.go:27]: My mistake. This field is not needed at all. The credential_id is in the instances block which has the right description.

**Review by spikewang — APPROVED** [review]
Thanks @Alshadex

**Pull request merged**